### PR TITLE
[fb62merge] Make RNTester launch without any redbox errors

### PR DESCRIPTION
**Pull request by alloy**
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [x] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

* re-applies some changes previously made but got lost in the merge
* removes a few `.macos.js` files that are no longer needed
* adds a native fix for code that would fail at runtime
* adds a few misc new fixes

## Test Plan

Screenshot 2020-05-18 at 14 42 39

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-macos/pull/372)

### CODE REVIEW

File: Libraries/Text/TextInput/Singleline/RCTUITextField.m [hunk 1]
```diff
@@ -247,7 +247,7 @@ - (void)_updatePlaceholder
   self.attributedPlaceholder = [[NSAttributedString alloc] initWithString:self.placeholder ?: @""
                                                                attributes:[self _placeholderTextAttributes]];
 #else // [TODO(macOS ISS#2323203)
-  self.placeholderAttributedString = [[NSAttributedString alloc] initWithString:self.placeholder ?: @""
+  self.placeholderAttributedString = [[NSAttributedString alloc] initWithString:self.placeholderString ?: @""
```
Comment (alloy) [inline Libraries/Text/TextInput/Singleline/RCTUITextField.m:250]: So this one is interesting, I think the compiler thinks it’s available due to `RCTUITextField` implementing the `RCTBackedTextInputViewProtocol` protocol, which has a `placeholder` property [here](https://github.com/microsoft/react-native-macos/blob/a87bd69ed6fb0e7d411087c19eec40114c246419/Libraries/Text/TextInput/RCTBackedTextInputViewProtocol.h#L37). However, that seems to rely on `UITextField` actually having a `placeholder` property, whereas `NSTextField` has `placeholderString`, which is also used elsewhere.

However, `placeholder` is still being used [here](https://github.com/microsoft/react-native-macos/blob/a87bd69ed6fb0e7d411087c19eec40114c246419/Libraries/Text/TextInput/Singleline/RCTUITextField.m#L491) and was also used in 0.61; what gives?

File: Libraries/Components/ScrollView/ScrollView.js [hunk 2]
```diff
@@ -57,12 +57,9 @@ if (Platform.OS === 'android') {
   AndroidScrollView = ScrollViewNativeComponent;
   AndroidHorizontalScrollView = AndroidHorizontalScrollViewNativeComponent;
   AndroidHorizontalScrollContentView = AndroidHorizontalScrollContentViewNativeComponent;
-} else if (Platform.OS === 'ios') {
+} else {
```
Comment (tom-un) [inline Libraries/Components/ScrollView/ScrollView.js:60]: To maintain minimal delta's from upstream, I'd prefer this line to become `if (Platform.OS === 'ios' || Platform.OS === 'macos' /*TODO(macOS ISS#2323203)*/) {` and remove the surrounding changes.   Also, its possible the final `else` case is actually used by another platform, maybe react-native-web ?

File: Libraries/Text/TextInput/Singleline/RCTUITextField.m (hunk 1 above)
Comment (tom-un) [reply Libraries/Text/TextInput/Singleline/RCTUITextField.m:250]: In RCTUITextField there are `setPlaceholder:` and `placeholder` selectors defined.   The impls have #if/defs to make mac call super setPlaceholderString/placeholderString.   The setter was overridden upstream too.   Did they get removed in .62?

Comment (tom-un) [reply Libraries/Text/TextInput/Singleline/RCTUITextField.m:250]: Ya, it looks like the setPlaceholder: selector is still in the fork, but somehow during the merge our fork's addition of the getter `placeholder` got deleted.   It just needs to be restored to be after the `setPlaceholder:` in `RCTUITextField.m`:
```m
- (NSString*)placeholder // [TODO(macOS ISS#2323203)
{
#if !TARGET_OS_OSX 
  return super.placeholder;
#else
  return self.placeholderAttributedString.string ?: self.placeholderString;
#endif
} // ]TODO(macOS ISS#2323203)
```

Comment (alloy) [reply Libraries/Text/TextInput/Singleline/RCTUITextField.m:250]: Aha, the pieces fall into place 👍

File: Libraries/Lists/VirtualizedSectionList.js [hunk 3]
```diff
@@ -19,7 +19,6 @@ const invariant = require('invariant');
 
 import type {ViewToken} from './ViewabilityHelper';
 import type {
-  Props as VirtualizedListProps,
   SelectedRowIndexPathType, // TODO(macOS ISS#2323203)
 } from './VirtualizedList';
```
Comment (tom-un) [inline Libraries/Lists/VirtualizedSectionList.js:23]: eslint will probably want this to a single line now:
```
import type { SelectedRowIndexPathType } from './VirtualList';  // TODO(macOS ISS#2323203)
```

File: Libraries/Components/ScrollView/ScrollView.js (hunk 2 above)
Comment (alloy) [reply Libraries/Components/ScrollView/ScrollView.js:60]: Good catch, I was looking at a different upstream tree.

**Comment by alloy** [comment]
Done 👍

**Comment by anandrajeswaran** [comment]
Thanks @alloy !

**Pull request merged**